model/src/id/mod.rs

```diff
@@ -86,6 +87,31 @@ impl<T> Id<T> {
             value,
         }
     }
+    /// Create a new non-zero ID while checking the value.
+    ///
+    /// This is primarily useful in const contexts where you are passing a
+    /// hardcoded value.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use twilight_model::id::{marker::GenericMarker, Id};
+    ///
+    /// const ID: Id<GenericMarker> = Id::new_checked(123);
+    ///
+    /// println!("id: {}", ID);
+    /// ```
+    ///
+    /// # Panics
+    ///
+    /// Panics if the value is 0.
+    pub const fn new_checked(n: u64) -> Self {
```

Comment:
`_checked` never panics, that's why it's called checked, e.g. [usize::checked_add](https://doc.rust-lang.org/std/primitive.usize.html#method.checked_add).
I do however support changing `new` to panic and `new_checked` to return an Option.

Reply:
Especially since creating an Id of 0 is almost always a programing error